cmd/argocd/commands/login.go

```diff
@@ -346,10 +356,86 @@ func oauth2Login(
 	return tokenString, refreshToken
 }
 
+func oauth2LoginBrowserless(
```

Comment:
This is a pretty big function.... can it be broken down into smaller pieces, ideally things that can be unit-tested?